examples/04_New Models.ipynb

```diff
@@ -835,6 +835,190 @@
     "## Matrix Models"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "At this point in the notebook, we have seen a few different model types. In all of these models, we have used dictionaries to represent the states. For example, in the implementation of `ThrownObject_ST` above, see how `dx` is defined with a StateContainer dictionary. While all models can be constructed in this way, Matrix Models provide an advanced feature that is applicable and useful for some use-cases. \n",
+    "\n",
+    "Matrix models are a particular type of model that represent the state of the system using matrices instead of dictionaries. The provided model.StateContainer, InputContainer, and OutputContainer can be treated as dictionaries but use an underlying matrix. This is important for some applications like surrogate and machine-learned models where the state is represented by a tensor. In a matrix model, simulation functions propagate the state using the matrix form, preventing the inefficiency of having to convert to and from dictionaries."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we'll illustrate how to create a matrix model. We'll continue with our ThrownObject system, and create a model to simulate this using matrix notation (instead of dictionary notation as in the standard model, seen above in `ThrownObject_ST`). The implementation of the model is comparable to a standard model, except that it uses matrix operations within each function, as seen below. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, the necessary imports."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from progpy import PrognosticsModel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll start by defining our matrix model class. Similarly to the standard models above, we'll define keys for the inputs, states, outputs, and events, as well as default parameters. We'll also include methods for initialization, output, threshold met, and event state. All of this is the same as the standard implementation. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ThrownObject_MM(PrognosticsModel):\n",
```

Comment:
Do we have to define the full thing like this, or can we just import ThrownObject and use that as the base class for ThrownObject_MM and only redefine next_state?

Reply:
Good point, this feels unnecessary. Importing ThrownObject and redefining next_state is good. Alternatively, we could also create ThrownObject_MM as a subclass of the ThrownObject_ST model created above in this notebook (i.e. the standard state-transition version, no drag), so that it's building off of something the user has (presumably) just read. What do you think?

Reply:
Oh I like that. Let's build off of the earlier model.